processing/src/main/java/io/druid/query/filter/Filter.java

```diff
@@ -25,6 +25,53 @@
  */
 public interface Filter
 {
+  /**
+   * Get a bitmap index, indicating rows that match this filter.
+   *
+   * @param selector Object used to retrieve bitmap indexes
+   * @return A bitmap indicating rows that match this filter.
+   */
   public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector);
+
+
+  /**
+   * Get a ValueMatcher that applies this filter to row values.
+   *
+   * @param factory Object used to create ValueMatchers
+   * @return ValueMatcher that applies this filter to row values.
+   */
   public ValueMatcher makeMatcher(ValueMatcherFactory factory);
+
+
+  /**
+   * Indicates whether this filter can return a bitmap index for filtering, based on
+   * the information provided by the input BitmapIndexSelector.
+   *
+   * @param selector Object used to retrieve bitmap indexes
+   * @return true if this Filter can provide a bitmap index using the selector, false otherwise
+   */
+  public boolean supportsBitmapIndex(BitmapIndexSelector selector);
+
+
+  /**
+   * Get a ValueMatcher that applies this filter to row values.
+   *
+   * Unlike makeMatcher(ValueMatcherFactory), this method allows the Filter to utilize bitmap indexes.
+   *
+   * An implementation should either:
+   * - return a ValueMatcher that checks row values, using the provided ValueMatcherFactory
+   * - or, if possible, get a bitmap index for this filter using the BitmapIndexSelector, and
+   *   return a ValueMatcher that checks the current row offset, created using the bitmap index.
+   *
+   * @param selector Object used to retrieve bitmap indexes
+   * @param valueMatcherFactory Object used to create ValueMatchers
+   * @param rowOffsetMatcherFactory Object used to create RowOffsetMatchers
+   * @return ValueMatcher that applies this filter
+   */
+  public ValueMatcher makeMatcher(
```

Comment:
I've moved this more complex makeMatcher() interface to the BooleanFilter and removed it from Filter

The And/OrFilters still need access to both the offsetMatcherFactory and the regular valueMatcherFactory, since it's possible for their subfilters to individually support bitmap indexes (or not).